module/documents/actor/actor.mjs

```diff
@@ -1488,15 +1488,27 @@ export default class Actor5e extends Actor {
       if ( Number.isNumeric(abilityValue) ) parts.push(String(abilityValue / 100));
     }
 
-    // Create the d20 roll
-    const formula = parts.join(" + ");
-    return new CONFIG.Dice.D20Roll(formula, data, {
-      flavor: flavor ?? game.i18n.localize("DND5E.Initiative"),
-      advantageMode: advantageMode,
-      halflingLucky: flags.halflingLucky ?? false,
+    foundry.utils.mergeObject(options, {
+      flavor: options.flavor ?? game.i18n.localize("DND5E.Initiative"),
+      halflingLuck: flags.halflingLucky ?? false,
       critical: null,
       fumble: null
     });
+
+    /**
+     * A hook event that fires before initiative is rolled for an Actor.
+     * @function dnd5e.preRollInitiative
+     * @memberof hookEvents
+     * @param {Actor5e} actor                The Actor that is rolling initiative.
+     * @param {object} data                  The roll data.
+     * @param {string[]} parts               The roll parts.
+     * @param {D20RollConfiguration} config  Configuration data for the pending roll.
+     */
+    Hooks.callAll("dnd5e.preRollInitiative", this, data, parts, options);
```

Comment:
I don't think we can support cancelling the roll by returning `false` from a hook here because this workflow is integrated with core `Combat#rollInitiative`, which does not have any provision for cancelling a roll based on the return value of `Combatant#getInitiativeRoll`.

Reply:
That seems fine here. Other places where hooks can cancel rolls it is expected that modules would divert to an alternative, but for initiative everything eventually needs to end up back in the core initiative system.